compiler/rustc_infer/src/infer/canonical/query_response.rs

```diff
@@ -50,11 +50,11 @@ impl<'tcx> InferCtxt<'tcx> {
     /// - Finally, if any of the obligations result in a hard error,
     ///   then `Err(NoSolution)` is returned.
     #[instrument(skip(self, inference_vars, answer, fulfill_cx), level = "trace")]
-    pub fn make_canonicalized_query_response<T>(
+    pub fn make_canonicalized_query_response<T, E: FulfillmentErrorLike<'tcx>>(
         &self,
         inference_vars: CanonicalVarValues<'tcx>,
         answer: T,
-        fulfill_cx: &mut dyn TraitEngine<'tcx>,
+        fulfill_cx: &mut dyn TraitEngine<'tcx, E>,
```

Comment:
shouldn't this always use scrubbed errors? we never emit errors inside of a canonical query after all 🤔

Reply:
This was removed in a follow-up I believe

Reply:
Wait no, it's because `ScrubbedTraitError` is defined in a downstream crate (`rustc_trait_selection`). And pulling it up to `rustc_infer` is a bit annoying, because then the `impl FromFulfillmentError for OldSolverError` is foreign because that type technically comes from `rustc_data_structures`, but I guess I could do it.